src/widgets/mod.rs

```diff
@@ -87,36 +84,83 @@ impl<W> WidgetDataHolder<W, NoData> {
     }
 }
 
-impl<W, D> WidgetDataHolder<W, D>
+impl<W> WidgetDataHolderTrait for NoDataHolder<W> {
+    type Owner = W;
+    type Data = NoData;
+
+    fn data_mut(&mut self) -> &mut Self::Data {
+        &mut self._no_data
+    }
+    fn update(&mut self, _widget: &mut Self::Owner) {}
+    fn on_data_changed(
+        &mut self,
+        _callback: fn(&mut Self::Owner, &<Self::Data as WidgetData>::Data),
+    ) {
+    }
+}
+
+pub trait WidgetDataHolderTrait {
+    type Owner;
+    type Data: WidgetData;
+
+    fn data_mut(&mut self) -> &mut Self::Data;
+
+    fn update(&mut self, widget: &mut Self::Owner);
+
+    fn on_data_changed(
+        &mut self,
+        _callback: fn(&mut Self::Owner, &<Self::Data as WidgetData>::Data),
+    ) {
+    }
+}
+
+pub struct WidgetDataHolder<W, D>
```

Comment:
Multiply the size of your types with this easy trick! Implementors of `WIdgetDataHolderTrait` really must avoid `W`